snap-geotiff/src/main/java/org/esa/snap/dataio/geotiff/GeoTiffProductReader.java

```diff
@@ -469,7 +465,15 @@ private static boolean applyGeoCoding(TiffFileInfo info, TIFFImageMetadata metad
 
         boolean isGlobalShifted180 = false;
         if (info.containsField(GeoTIFFTagSet.TAG_MODEL_TIE_POINT)) {
-            double[] tiePoints = info.getField(GeoTIFFTagSet.TAG_MODEL_TIE_POINT).getAsDoubles();
+            TIFFField tiePointsField = info.getField(GeoTIFFTagSet.TAG_MODEL_TIE_POINT);
+            double[] tiePoints;
+            try{
+                tiePoints = tiePointsField.getAsDoubles();
+            }catch (Exception e){
```

Comment:
Actually, this should result in an IOException or IllegalFileFormatException. The specification requires Double values, not a String. The same for the other modifcation in this file and in TiffFileInfo.

If the changes are really necessary, these modifications can be extracted to a new method getAsDoubles(TIFFField).